packit/utils/obs_helper.py

```diff
@@ -0,0 +1,452 @@
+        if arch in ("s390x", "aarch64", "ppc64le"):
+            postfix = {"s390x": "zSystem", "aarch64": "ARM", "ppc64le": "PowerPC"}[arch]
+            return [
+                XmlPathEntry(
+                    project=f"openSUSE:Factory:{postfix}",
+                    repository="standard",
+                ),
+            ]
+
+    raise ValueError(f"No preset available for {distro=}, {version=}, {arch=}")
+
+
+def targets_to_project_meta(
+    targets: list[str],
+    owner: str,
+    project_name: str,
+    description: Optional[str] = None,
+) -> ET.Element:
+    """Converts the list of packit targets (like `fedora-rawhide`) to a project
+    meta xml configuration for the respective project in OBS.
+
+    """
+    repos: list[OBSRepository] = []
+    for target in targets:
+        path = target_to_path(target)
+        arch = target.split("-")[-1]
+        name = target
+        added = False
+
+        for ind, repo in enumerate(repos):
+            if repo.path == path:
+                repos[ind] = OBSRepository(
+                    name=f"{repo.name}-{arch}",
+                    path=path,
+                    arch=[*repo.arch, arch],
+                )
+                added = True
+
+        if not added:
+            repos.append(OBSRepository(name=name, path=path, arch=[arch]))
+
+    root = ET.Element("project")
+    root.attrib["name"] = project_name
+
+    (title_elem := ET.Element("title")).text = "Packit project"
+    (descr_elem := ET.Element("description")).text = description or ""
+    (person_elem := ET.Element("person")).attrib["userid"] = owner
+    person_elem.attrib["role"] = "maintainer"
+
+    for elem in (title_elem, descr_elem, person_elem):
+        root.append(elem)
+
+    for repo in repos:
+        (repo_elem := ET.Element("repository")).attrib["name"] = repo.name
+
+        for path_entry in repo.path:
+            (path_elem := ET.Element("path")).attrib["project"] = path_entry.project
+            path_elem.attrib["repository"] = path_entry.repository
+            repo_elem.append(path_elem)
+
+        for arch in repo.arch:
+            (arch_elem := ET.Element("arch")).text = arch
+            repo_elem.append(arch_elem)
+
+        root.append(repo_elem)
+
+    return root
+
+
+def parse_changelog_subject(line: str):
+    date: list[str]
+    author: list[str] = []
+    email: str = ""
+    version: str = ""
+
+    toks = [tok.strip() for tok in line.split(" ") if tok.strip()]
+
+    toks.pop(0)  # Remove leading "*"
+    date = toks[:4]
+
+    del toks[:4]
+    for index, element in enumerate(toks):
+        if not element.startswith("<"):
+            author.append(element)
+        else:
+            email += element
+            del toks[: index + 1]
+            break
+
+    if toks and toks[0] == "-":
+        toks.pop(0)
+
+    if toks:
+        version = toks[0]
+
+    return date, author, email, version
+
+
+def format_changelog(changelog: list[str]) -> list[str]:
```

Comment:
Suggestion: convert the signature as follows
```suggestion
def format_changelog(changelog: str) -> str:
```
atm you're using `format_changelog` everywhere as follows:
```Python
changes_file_content = "\n".join(
    format_changelog(changes_file_content.splitlines()),
)
```
this is usually an indicator that the splitting and joining part should be part of your function and not passed as an exercise to the API consumer.

Reply:
Improved the input and return types of `format_changelog`